### PR TITLE
[token-CLI] Add command_initialize_group to Token Cli

### PR DESCRIPTION
**Pull request by tonton-sol**
I have added the necessary clap commands and functions to drive `token_group_initialize_with_rent_transfer()` from the Token CLI.  I am running into error 0xc `InvalidInstruction` and cannot find what is causing this. Is there anything obvious that I cannot see? #5937

Thank you!

Edit: To recreate this error, run
```
spl-token --program-id TokenzQdBNbLqP5VEhdkAS6EPFLC1PHnBqCXEpPxuEb create-token --enable-group
```
then I initialize the group using the mint address created
```
spl-token initialize-group A6iv673L7PEZDxYDwQbMZsgmS5bGzExavADQSAcKA3Tt 3   
```

### CODE REVIEW

**Comment by tonton-sol** [comment]
@buffalojoec please let me know what you think!

**Comment by buffalojoec** [comment]
> I am running into error 0xc InvalidInstruction and cannot find what is causing this. Is there anything obvious that I cannot see?

Are you running this test with Token2022's feature flag enabled for `token-group`?

**Comment by tonton-sol** [comment]
Yes I believe it is enabled. And the error comes before the token-group processor checks if it is enabled.

**Comment by buffalojoec** [comment]
> Yes I believe it is enabled. And the error comes before the token-group processor checks if it is enabled.

And you're using the CLI from the repo? `target/debug/spl-token`?

**Comment by tonton-sol** [comment]
Yes, and I can see the new clap subcommand I added.

**Comment by buffalojoec** [comment]
The subcommand is just on the CLI side. We'll have to configure the tests to use Token2022's program `.so` file with the features enabled.

To test it manually, you can do:

```
cargo build-sbf --manifest-path token/program-2022/Cargo.toml --features token-group zk-ops
solana-test-validator --bpf-program TokenzQdBNbLqP5VEhdkAS6EPFLC1PHnBqCXEpPxuEb target/deploy/spl_token_2022.so
```

And then we'll just have to configure that same BPF program override in the test validator config of the tests. This might be outside the scope of this PR though.

If you manage to crack it, great! If not, we can wait on this and rebase after I add that support.

**Comment by tonton-sol** [comment]
I had a fundamental misunderstanding lol. The deployed devnet token-22 program doesn't have the token-group enabled. I followed the steps for testing the cli and enabled it correctly. I also wrote a test that simply creates the mint and initializes it. The test passes. What else should I add to the test? Maybe check the max_size? Since the `GroupPointer` struct does not have a max_size field, would I need to read it directly from the raw data?

**Comment by tonton-sol** [comment]
Actually I think this can be done with the `spl-token-group-interface`! Would it make sense to roll `command_update_max_size` into this pr? For example it seems the metadata test includes multiple commands.

**Comment by buffalojoec** [comment]
> Actually I think this can be done with the `spl-token-group-interface`! Would it make sense to roll `command_update_max_size` into this pr? For example it seems the metadata test includes multiple commands.

Maybe just try fetching the mint and unpacking `TokenGroup` and checking the values for things like update authority/mint?

**Comment by tonton-sol** [comment]
Okay I believe this is ready to go!

File: token/cli/src/clap_app.rs [hunk 1]
```diff
@@ -962,6 +963,52 @@ pub fn app<'a, 'b>(
...
+                )
+                .arg(
+                        Arg::with_name("max_size")
+                        .validator(is_amount)
+                        .value_name("MAX_SIZE")
+                        .takes_value(true)
+                        .required(true)
+                        .index(2)
+                        .help("The number of members in the group."),
+                    )
```
Comment (buffalojoec) [inline token/cli/src/clap_app.rs:986]: Just an idea here, but what if we allowed for the keyword `max` here, and we'd default that to `U32::MAX`?

**Review by buffalojoec — COMMENTED** [review]
Looks good! See what you think on my "max size" suggestion, but besides that I'm ready to stamp this one.

File: token/cli/src/clap_app.rs (hunk 1 above)
Comment (tonton-sol) [reply token/cli/src/clap_app.rs:986]: I think to do this we would need to create a new validator in the solana-clap-utils like `is_amount_or_max`. I am also not sure how common of a use case that would be. I think its good as is 👍 

After this, I will work my way through the rest of the commands as separate PRs. 😎

**Review by buffalojoec — APPROVED** [review]
Thanks!
group hug

**Pull request merged**